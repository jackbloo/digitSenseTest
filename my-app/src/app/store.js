import { configureStore } from '@reduxjs/toolkit';
import dataReducer from '../reducer/data/reducer';

export const store = configureStore({
  reducer: {
    dataPicker: dataReducer
  },
});
