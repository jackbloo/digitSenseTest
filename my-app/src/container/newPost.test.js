import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux';
import { store } from '../app/store';
import NewPost from './newPost'
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'


describe('Unit Test for NewPost', () => {
    const history=createMemoryHistory()
    test('Should be in the document', () => {
     render(
          <Provider store={store} >
              <Router navigator={history} location={history.location}>
            <NewPost />
            </Router>
          </Provider>
        );
      const card = screen.getByTestId('new-post-container')
      expect(card).toBeInTheDocument();
      });

      test('Should match the snapshot', () => {
        render(
             <Provider store={store} >
              <Router navigator={history} location={history.location}>
            <NewPost />
            </Router>
             </Provider>
           );
         const card = screen.getByTestId('new-post-container')
         expect(card).toMatchSnapshot()
         });
})

