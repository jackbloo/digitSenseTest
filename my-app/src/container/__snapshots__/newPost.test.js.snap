// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for Card Should match the snapshot 1`] = `
<div
  class="sc-bdvvtL dHBknt"
  data-testid="new-post-container"
>
  <div
    class="sc-gsDKAQ cBbBvl"
  >
    New Post
  </div>
  <div
    class="sc-furwcr bhzqyp"
  >
    Back to Home
  </div>
  <div
    class="sc-dkPtRN jJpDfM"
  >
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Title
      </div>
      <input
        class="sc-eCImPb dAIa-DJ"
        placeholder="Insert Title"
        value=""
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Content
      </div>
      <textarea
        class="sc-jRQBWg ilgbxP"
        placeholder="Insert Content"
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Author
      </div>
      <input
        class="sc-eCImPb dAIa-DJ"
        placeholder="Insert Author"
        value=""
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-iCfMLu cTEiGp"
      >
        Submit
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit Test for NewPost Should match the snapshot 1`] = `
<div
  class="sc-bdvvtL dHBknt"
  data-testid="new-post-container"
>
  <div
    class="sc-gsDKAQ cBbBvl"
  >
    New Post
  </div>
  <div
    class="sc-furwcr bhzqyp"
  >
    Back to Home
  </div>
  <div
    class="sc-dkPtRN jJpDfM"
  >
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Title
      </div>
      <input
        class="sc-eCImPb dAIa-DJ"
        placeholder="Insert Title"
        value=""
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Content
      </div>
      <textarea
        class="sc-jRQBWg ilgbxP"
        placeholder="Insert Content"
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-gKclnd eXedrW"
      >
        Author
      </div>
      <input
        class="sc-eCImPb dAIa-DJ"
        placeholder="Insert Author"
        value=""
      />
    </div>
    <div
      class="sc-hKwDye hJIobn"
    >
      <div
        class="sc-iCfMLu cTEiGp"
      >
        Submit
      </div>
    </div>
  </div>
</div>
`;
